index.js

```diff
@@ -1,34 +1,31 @@
 const dotenv = require('dotenv');
 dotenv.config();
 process.env.MONGO_URL;
-
-const { MongoClient } = require('mongodb');
-
 const express = require('express');
-const path = require('path');
-
 const port = process.env.PORT || 8080;
 const app = express();
+const path = require('path');
+const mongoose = require('mongoose');
+const authRoute = require('./lib/routes/auth');
+const postRoute = require('./lib/routes/posts');
 
 app.use(express.static(path.join(__dirname, 'client/build')));
 
-app.get('/*', function (req, res) {
-  res.sendFile(path.join(__dirname, 'client/build', 'index.html'));
+app.get('*', (request, response) => {
+  response.sendFile(path.join(__dirname, 'client/build/index.html'));
 });
 
-const client = new MongoClient(process.env.MONGO_URL, {
-  useNewUrlParser: true,
-  useUnifiedTopology: true,
-});
+mongoose.connect(
```

Comment:
Make sure that you start the webserver after your DB is connected as you did with the pure MongoDB solution.
Otherwise, it's possible that the server starts without a connection to the DB.